src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/DefaultRazorProjectFileSystem.cs

```diff
@@ -103,6 +103,14 @@ protected override string NormalizeAndEnsureValidPath(string path)
             return normalizedPath;
         }
 
+        // This might be an absolute path rooted outside of the project root. In that case,
```

Comment:
Do we need to consider long path syntax here? For example: `\\?\...`? If so should we do this check before the one at lines 100-101? The code as written should still work but seems more intentional to do this check first.

Reply:
Possibly. The [original code](https://github.com/dotnet/razor/blob/97c495207d68dc083874a5c1c198378dd9b449be/src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/DefaultRazorProjectFileSystem.cs#L72-L102) never checked for long paths, but that doesn't mean it isn't a good idea.

I'll move this check early and make it Windows-only. It turns out that the semantics of this method are a bit sketchy and allows `path` to be a fully-qualified path on Unix platforms.